### PR TITLE
[SPARK-20916][SQL] Improve error message for unaliased subqueries in FROM clause

### PR DESCRIPTION
**Pull request by viirya**
## What changes were proposed in this pull request?

We changed the parser to reject unaliased subqueries in the FROM clause in SPARK-20690. However, the error message that we now give isn't very helpful:

    scala> sql("""SELECT x FROM (SELECT 1 AS x)""")
    org.apache.spark.sql.catalyst.parser.ParseException:
    mismatched input 'FROM' expecting {<EOF>, 'WHERE', 'GROUP', 'ORDER', 'HAVING', 'LIMIT', 'LATERAL', 'WINDOW', 'UNION', 'EXCEPT', 'MINUS', 'INTERSECT', 'SORT', 'CLUSTER', 'DISTRIBUTE'}(line 1, pos 9)

We should modify the parser to throw a more clear error for such queries:

    scala> sql("""SELECT x FROM (SELECT 1 AS x)""")
    org.apache.spark.sql.catalyst.parser.ParseException:
    The unaliased subqueries in the FROM clause are not supported.(line 1, pos 14)

## How was this patch tested?

Modified existing tests to reflect this change.

### CODE REVIEW

**Comment by viirya** [comment]
cc @JoshRosen @cloud-fan @hvanhovell @gatorsmile

**Comment by maropu** [comment]
Better to add tests in `SQLQueryTestSuite`?

**Comment by viirya** [comment]
@maropu Sure. Thanks.

File: sql/core/src/test/resources/sql-tests/inputs/aliased-subquery-in-from.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+-- Aliased subqueries in FROM clause
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/aliased-subquery-in-from.sql]: Could you move the test case `sql-tests/inputs/aliased-subquery-in-from.sql` to `sql-tests/inputs/subquery/subquery-in-from.sql`?

**Review by gatorsmile — APPROVED** [review]
LGTM except one comment

File: sql/core/src/test/resources/sql-tests/inputs/aliased-subquery-in-from.sql (hunk 1 above)
Comment (viirya) [inline sql/core/src/test/resources/sql-tests/inputs/aliased-subquery-in-from.sql]: Sure.

**Comment by viirya** [comment]
retest this please.

**Comment by cloud-fan** [comment]
LGTM, merging to master!

**Pull request closed without merging**